tracee-rules/benchmark/benchmark_reference_test.go

```diff
@@ -0,0 +1,160 @@
+// Reference: https://github.com/open-policy-agent/golang-opa-wasm/blob/8041bec28e319aad31fd4632bae46e47727f3ddc/opa/opa_test.go#L154-L216
```

Comment:
What is the difference between benchmark_reference_test.go and benchmark_test.go? Do we need both Go test files?

Reply:
`benchmark_reference_test.go` is a mostly standalone benchmark that shows performance between WASM/Rego without any additional tracee dependencies. I added it in so as to share with the OPA team if they need to evaluate it. 

We can put them in the same file if that's better.

Reply:
The goal is to have reliable and useful benchmarks that can help us make decisions, sharing them externally is desirable side-effect but shouldn't drive the design of the tests. To address the specific points:
- I don't see a point in benchmarking both sample signatures and a real signatures. The content and name of the signature is meaningless but it needs to so something typical like 3-5 lines of assertions on input at least. Any of the existing signatures would be ok.
- We shouldn't use or test the "golang-opa-wasm" library, as suggested by the maintainers
- I agree it's better to make changes in the upstream code as opposed to rewriting the tested code in the test
- However - we shouldn't automatically upstream all the possible variations we benchmark (that's why we test them first). For specific cases, like WASM, sounds like a quick win we would want to have anyway, so yeah. For other cases, maybe we can consider upstreaming support for user-provided flags if we want it to be more generic and not necessarily in response to a benchmark (won't help all variations though).
- If we want to keep the "reference" benchmarks just for sharing externally, we could move it to personal repo (they don't really relate to tracee)